### ISSUE TITLE
Define Custom Keybindings

### ISSUE THREAD

**Issue by sickerin** [joouha/euporie#27]
Hi @joouha 
Is there a way to define custom keybindings? Most of the keybindings (Ctrl+N, etc) conflict with emacs keybindings/ terminals. So for the stuff that I rarely use, I'd probably turn them off.

Right now it's not possible to load some config right? I presume I need to change https://github.com/joouha/euporie/blob/main/euporie/key_binding/bindings/micro.py

Maybe having an easy method for editing custom keybindings could be an interesting addition. I might look into this.

**Comment by joouha** [comment]
Hi @sickerin,

This is not possible in `v1.6.1`, but it will be possible in `v2.0.0` when it's released.

I've already implemented this in the `dev` branch if you want to check it out.

To configure the key-bindings in `euporie-notebook`, you'll need to add something like the following to `$XDG_CONFIG_HOME/euporie/config.json`:

```json
{
  "notebook": {
    "key_bindings": {
      "euporie.notebook.app.NotebookApp": {
        "new-notebook": [],
        "quit": ["c-q", "c-p"],
      }
    },
  }
}
```

This will disable the <kbd>Ctrl+N</kbd> key-binding for the `new-notebook` command, and adds an additional key-binding <kbd>Ctrl+P</kbd> for the `quit` command.

A key-binding editor dialog would be a nice addition, as editing the config file is a bit fiddly.

**Comment by sickerin** [comment]
Awesome, will take a look!

**Comment by joouha** [comment]
I've released v2.0.0, which now supports custom key-bindings:
https://euporie.readthedocs.io/en/latest/pages/keybindings.html#custom-key-bindings
